frontend/src/hooks/useLocalStorage.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+
+export function useLocalStorage<T>(key: string, defaultValue: T): [T, (value: T) => void] {
+  const [storedValue, setStoredValue] = useState(defaultValue)
+
+  useEffect(() => {
+    const item = localStorage.getItem(key)
+
+    if (!item) {
+      localStorage.setItem(key, JSON.stringify(defaultValue))
+    }
+
+    setStoredValue(item ? JSON.parse(item) : defaultValue)
+
+    function handler(e: StorageEvent) {
+      if (e.key !== key) return
+
+      const lsi = localStorage.getItem(key)
+      setStoredValue(JSON.parse(lsi ?? ''))
```

Comment:
what is lsi?

Reply:
This is just a already made hook that I took from internet, because it's so popular. But I suppose lsi comes from local storage item. I can change it in something more clear